SPMeta2/SPMeta2.CSOM/ModelHandlers/WebModelHandler.cs

```diff
@@ -397,9 +398,22 @@ private static void MapProperties(Web web, WebDefinition webModel)
             if (webModel.IndexedPropertyKeys.Any())
             {
                 var props = web.AllProperties;
-                // TODO Not sure if property name should be hardcoded in here and if web.Update needs to be called here
-                props["vti_indexedpropertykeys"] = GetEncodedValueForSearchIndexProperty(webModel.IndexedPropertyKeys);
+               
+                // If other property keys have been set before, overwriting them would be bad
+                if (props["vti_indexedpropertykeys"] == null || props["vti_indexedpropertykeys"].ToString() == string.Empty)
+                {
+                    props["vti_indexedpropertykeys"] = GetEncodedValueForSearchIndexProperty(webModel.IndexedPropertyKeys);
+                }
+                else
+                {
+                    var indexPropertyValue = props["vti_indexedpropertykeys"].ToString();
+
+                    var indexList = GetDecodeValueForSearchIndexProperty(indexPropertyValue);
```

Comment:
You're absolutely right. Looking forward to those ConvertUtils and if I remember correctly reSP suggested something similar for checking for null and string.Empty.

Still a lot of things I got to learn and I'm improving with every pull request ;)

Will you make the changes or can I add an other change to this pull request?